### ISSUE TITLE
https://games.arantius.com/game.86.8+planet.html

### ISSUE THREAD

**Issue by cmlad26** [ruffle-rs/ruffle#1991]
# Error Info
Error name: Error
Error message: panicked at 'called `Option::unwrap()` on a `None` value', core/src/display_object/container.rs:780:18
Error stack:
```
Error: panicked at 'called `Option::unwrap()` on a `None` value', core/src/display_object/container.rs:780:18
    at Object.Y (https://static.arantius.com/ruffle/core.ruffle.dd338b559c70010995b3.js:1:20416)
    at __wbg_new_df6e6ab7a65c4c4d (https://static.arantius.com/ruffle/ruffle.js:1:35177)
    at https://static.arantius.com/ruffle/db77e9b3a74d57a3cbd2.module.wasm:wasm-function[1074]:0x2158e4
    at https://static.arantius.com/ruffle/db77e9b3a74d57a3cbd2.module.wasm:wasm-function[1397]:0x254528
    at https://static.arantius.com/ruffle/db77e9b3a74d57a3cbd2.module.wasm:wasm-function[3731]:0x3093cc
    at https://static.arantius.com/ruffle/db77e9b3a74d57a3cbd2.module.wasm:wasm-function[2936]:0x2eb05c
    at https://static.arantius.com/ruffle/db77e9b3a74d57a3cbd2.module.wasm:wasm-function[4026]:0x30e003
    at https://static.arantius.com/ruffle/db77e9b3a74d57a3cbd2.module.wasm:wasm-function[4056]:0x30e566
    at https://static.arantius.com/ruffle/db77e9b3a74d57a3cbd2.module.wasm:wasm-function[3932]:0x30cc09
    at https://static.arantius.com/ruffle/db77e9b3a74d57a3cbd2.module.wasm:wasm-function[4057]:0x30e59c
```

# Player Info
Allows script access: false
Player type: Object
SWF URL: //static.arantius.com/games/8planet.swf
Param movie: //static.arantius.com/games/8planet.swf
Param quality: high
Param bgcolor: #FFFFFF
Attribute 0: undefined
Attribute 1: undefined
Attribute 2: undefined
Attribute 3: undefined
Attribute 4: undefined

# Page Info
Page URL: https://games.arantius.com/game.86.8+planet.html

# Browser Info
Useragent: Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.88 Safari/537.36
OS: Win32

# Ruffle Info
Version: 0.1.0
Name: nightly 2020-12-12
Channel: nightly
Built: 2020-12-12T00:33:23.026Z
Commit: 4358e52af746929abcbbd5f05e577b98a0992fdd

**Comment by Toad06** [comment]
Thanks for your report.

This seems to be the same symptom as #1671 and #1942. In `core/src/display_object/container.rs`, changing this:
```
if !matches!(child.object2(), Avm2Value::Undefined) {
	child.set_parent(context.gc_context, None);
}
```
to:
`child.set_parent(context.gc_context, None);`

prevents the crash from happening, though unexpected things could still happen, especially when you're close to clear the level.

**Comment by xTibor** [comment]
"The Binding of Isaac: Wrath of the Lamb" also triggers this panic after a few seconds in-game. (Commercial game, I cannot share SWF files.)

**Comment by Herschel** [comment]
Fixed by #2228.
